.gitignore

```diff
@@ -1,27 +1,106 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# I don't konw.
 _trial_temp
-build
-build-pypy
-MANIFEST
-dist
 apidocs
 *,cover
 .testrepository
-*.pyc
-*.pyd
-*.pyo
-*.so
 *~
-*.swp
-*.swh
-*.swn
-*.swo
+*.sw[phno]
 docs/tutorial/index.html
 dulwich.egg-info/
+docs/api/*.txt
+dulwich.dist-info
+
+# Unit test / coverage reports
+htmlcov/
 .tox/
+.coverage
+.coverage.*
+.cache
+nosetests.xml
```

Comment:
Dulwich doesn't use nosetest, hypothesis or pytest.

Reply:
We will use Pytest and I will add it later